test/integration/thermal_system.cc

```diff
@@ -169,3 +174,67 @@ TEST_F(ThermalTest, TemperatureComponent)
   EXPECT_TRUE(heatSignatures[heatSignatureSphereVisual2].find(
         heatSignatureTestResource) != std::string::npos);
 }
+
+/////////////////////////////////////////////////
+TEST_F(ThermalTest, ThermalSensorSystem)
+{
+  // Start server
+  ServerConfig serverConfig;
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
+      "/test/worlds/thermal.sdf");
+
+  Server server(serverConfig);
+  EXPECT_FALSE(server.Running());
+  EXPECT_FALSE(*server.Running(0));
+
+  // Create a system that checks for thermal component
+  test::Relay testSystem;
+
+  double resolution = 0;
+  double minTemp = std::numeric_limits<double>::max();
+  double maxTemp = 0u;
+  std::string name;
+  sdf::Sensor sensorSdf;
+  testSystem.OnUpdate([&](const gazebo::UpdateInfo &,
+    gazebo::EntityComponentManager &_ecm)
+    {
+      _ecm.Each<components::ThermalCamera, components::Name>(
+          [&](const ignition::gazebo::Entity &_id,
+              const components::ThermalCamera *_sensor,
+              const components::Name *_name) -> bool
+          {
+            // store temperature data
+            sensorSdf = _sensor->Data();
+            name = _name->Data();
+
+            auto resolutionComp =
+                _ecm.Component<components::TemperatureLinearResolution>(
+                _id);
+            EXPECT_NE(nullptr, resolutionComp);
+            resolution = resolutionComp->Data();
+
+            auto temperatureRangeComp =
+                _ecm.Component<components::TemperatureRange>(_id);
+            EXPECT_NE(nullptr, temperatureRangeComp);
+            auto info = temperatureRangeComp->Data();
+            minTemp = info.min.Kelvin();
+            maxTemp = info.max.Kelvin();
+            return true;
+          });
+
+    });
+  server.AddSystem(testSystem.systemPtr);
+
+  // Run server
+  server.Run(true, 1, false);
+
+  EXPECT_EQ("thermal_camera_8bit", name);
+  const sdf::Camera *cameraSdf = sensorSdf.CameraSensor();
+  ASSERT_NE(nullptr, cameraSdf);
```

Comment:
It might be worth checking that `sensorSdf.CameraSensor()` returns a `nullptr` before calling `server.Run`

Reply:
I found that I had to run one iteration first in order for sdf to be loaded otherwise the sdf will be null.